tests/console/yast2_ftp.pm

```diff
@@ -26,108 +27,95 @@ sub run() {
+    send_key 'alt-s';                                                              # select umask for authenticated users
+    type_string '0022';                                                            # set 755
+    assert_screen 'ftp_umask_value';                                               # check umask value
+    wait_screen_change { send_key 'alt-y' };                                       # give a new directory for anonymous users
+    send_key_until_needlematch 'yast2_tftp_empty_anon_dir', 'backspace';
     type_string '/srv/ftp/anonymous';
-    send_key 'alt-t';                              # give a new directory for authenticated users
+    send_key 'alt-t';                                                              # give a new directory for authenticated users
     wait_still_screen 1;
     type_string '/srv/ftp/authenticated';
-    assert_screen 'yast2_ftp_general_directories';    # check new directories for ftp users
+    assert_screen 'yast2_ftp_general_directories';                                 # check new directories for ftp users
     send_key 'alt-o';
     assert_screen 'yast2_ftp_directory_browse';
     send_key 'alt-c';
 
     # Performance
     send_key_until_needlematch 'yast2_tftp_general_selected', 'shift-tab';
-    send_key 'down';
-    wait_still_screen 1;
-    send_key 'ret';
-    wait_still_screen 1;
+    wait_screen_change { send_key 'down' };
+    wait_screen_change { send_key 'ret' };
     send_key 'alt-m';
-    for (1 .. 5) { send_key 'down'; }
-    send_key 'alt-e';    # change max client for one IP
-    for (1 .. 4) { send_key 'up'; }
-    send_key 'alt-x';    # change max clients to 20
-    for (1 .. 11) { send_key 'up'; }
-    send_key 'alt-l';    # change local max rate to 100 kb/s
-    for (1 .. 20) { send_key 'up'; }
-    send_key 'alt-r';    # change anonymous max rate to 50 kb/s
-    for (1 .. 10) { send_key 'up'; }
-    assert_screen 'yast2_ftp_performance-settings';    # check performance settings
+    type_string_slow "10\n";
+    send_key 'alt-e';                                                              # change max client for one IP
+    type_string_slow "7\n";
+    send_key 'alt-x';                                                              # change max clients to 20
+    type_string_slow "20\n";
+    send_key 'alt-l';                                                              # change local max rate to 100 kb/s
+    type_string_slow "100\n";
+    send_key 'alt-r';                                                              # change anonymous max rate to 50 kb/s
+    type_string_slow "50\n";
+    assert_screen 'yast2_ftp_performance-settings';                                # check performance settings
 
     # Authentication
     send_key_until_needlematch 'yast2_tftp_performance_selected', 'shift-tab';
-    send_key 'down';
+    wait_screen_change { send_key 'down' };
     wait_still_screen 1;
```

Comment:
that looks a bit redundant now. We should only need one of wait_screen_change/wait_still_screen